File: CloudStoragePlatform.Core/Domain/Entities/Metadata.cs
using CloudStoragePlatform.Core.Domain.IdentityEntites;
using CloudStoragePlatform.Core.DTO;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CloudStoragePlatform.Core.Domain.Entities
{
    public class Metadata
    {
        [Key]
        public Guid MetadataId { get; set; }
        public virtual File? File { get; set; }
        public virtual Folder? Folder { get; set; }

        public DateTime? PreviousRenameDate { get; set; }
        public int RenameCount { get; set; } = 0;

        public string? PreviousPath { get; set; }
        public DateTime? PreviousMoveDate { get; set; }
        public int MoveCount { get; set; } = 0;

        public DateTime? LastOpened { get; set; }
        public int OpenCount { get; set; } = 0;

        public int ShareCount { get; set; } = 0;
        public virtual ApplicationUser User { get; set; }
        public Guid UserId { get; set; }

        /// <summary>
        /// in MegaBytes
        /// </summary>

        public FileOrFolderMetadataResponse ToMetadataResponse() 
        {
            var response = new FileOrFolderMetadataResponse()
            {
                MetadataId = MetadataId,
                PreviousRenameDate = PreviousRenameDate,
                RenameCount = RenameCount,
                PreviousMoveDate = PreviousMoveDate,
                LastOpened = LastOpened,
                PreviousPath = PreviousPath,
                ShareCount = ShareCount,
                MoveCount = MoveCount,
                OpenCount = OpenCount,
            };
            return response;
        }
    }
}

File: Cloud Storage Platform/AngularApplication/CloudStoragePlatformUI/src/app/models/Metadata.ts
export class Metadata {
  parentFolderName: string;
  subFoldersCount: number;
  subFilesCount: number;
  creationDate: Date | null | string;
  metadataId: string;
  previousReplacementDate: Date | null | string;
  replaceCount: number;
  previousRenameDate: Date | null | string;
  renameCount: number;
  previousPath: string | null;
  previousMoveDate: Date | null | string;
  moveCount: number;
  lastOpened: Date | null | string;
  openCount: number;
  shareCount: number;
  size: number;

  constructor(data: Partial<Metadata> = {}) {
    this.parentFolderName = data.parentFolderName ?? "";
    this.subFoldersCount = data.subFoldersCount ?? 0;
    this.subFilesCount = data.subFilesCount ?? 0;
    this.creationDate = data.creationDate ? new Date(data.creationDate) : null;
    this.metadataId = data.metadataId ?? "";
    this.previousReplacementDate = data.previousReplacementDate ? new Date(data.previousReplacementDate) : null;
    this.replaceCount = data.replaceCount ?? 0;
    this.previousRenameDate = data.previousRenameDate ? new Date(data.previousRenameDate) : null;
    this.renameCount = data.renameCount ?? 0;
    this.previousPath = data.previousPath ?? null;
    this.previousMoveDate = data.previousMoveDate ? new Date(data.previousMoveDate) : null;
    this.moveCount = data.moveCount ?? 0;
    this.lastOpened = data.lastOpened ? new Date(data.lastOpened) : null;
    this.openCount = data.openCount ?? 0;
    this.shareCount = data.shareCount ?? 0;
    this.size = data.size ?? 0;
  }
}
